cypress/integration/chi-js/popover.spec.js

```diff
@@ -0,0 +1,107 @@
+describe('Popover', function() {
+  before(() => {
+    cy.visit('/tests/js/popover.html');
+    cy.fixture('chidata.json').then(data => {
+      this.chidata = data;
+    });
+  });
+  describe('Popover should be shown  when  button  clicked ', function() {
+    it('Popover should shown on button clicked', () => {
+      cy.get('[data-cy="show-popover"]')
+        .click()
+        .find('+ .chi-popover')
+        .should('be.visible')
+        .should('have.class', '-active');
+    });
+    it('Popover should hide  on button clicked twice', () => {
+      cy.get('[data-target="#popover-in-html"]')
+        .click()
+        .find('+ .chi-popover')
+        .should('be.not.visible');
+    });
+  });
+
+  describe('Popover Positioning and arrow should be matched ', () => {
+    it('Popover Positioning and arrow should work in accordance', () => {
+      this.chidata.popperPositions.forEach(position => {
+        const getValue = `[data-cy="test-more-${position}"]`;
+        cy.get(getValue)
+          .find('button.chi-button')
+          .should('match', `[data-position="${position}"]`)
+          .find('+ .chi-popover')
+          .should('match', `[x-placement="${position}"]`);
+      });
+    });
+  });
+
+  describe('Autohide and NoAutohide should work', function() {
+    it('popover should hide when click out side when preventAutoHide set to false', () => {
+      cy.get('[data-cy="auto-hide"]')
+        .find('[data-cy="popover-6"]')
+        .click()
+        .parent()
+        .find('.chi-popover#chi-popover-13')
+         .should('have.class', '-active')
+         .click(500, 0, {force: true})
+         .find('.chi-popover#chi-popover-13')
+        .should('not.have.class', '-active');
+    });
+
+    it('popover should not hide when click out side when preventAutoHide set to true', () => {
+      cy.get('[data-cy="auto-hide"]')
+        .find('[data-cy="popover-7"]')
+        .click()
+        .parent()
+        .find('.chi-popover#chi-popover-14')
+        .should('have.class', '-active')
+        .click(500, 0, {force: true})
+        .get('.parent')
+        .find('.chi-popover#chi-popover-14')
+        .should('have.class', '-active')
+        .and('match', `[aria-hidden="false"]`);
+    });
+  });
+
+  describe('Hide, Toggle, Show events should work ', function() {
```

Comment:
```suggestion
  describe('Hide, Toggle, Show methods should work ', function() {
```

Reply:
The click events call the .show(), .hide(), .toggle() methods.
Regarding events: Can we add test cases to check event emission of:
**chi.popover.show
chi.popover.hide** ?
Popover is supposed to emit these events when show and hide are triggered. Would be nice to test that as well.